coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,375 @@
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long lines of code and too large files result in code difficult to
+        read, understand and maintain.
+        """
+        fix_suggestions = """
+        Length issues can be fixed by writing shorter lines of code (splitting
+        long lines into multiple shorter lines); writing shorter files
+        (splitting files into modules, writing shorter methods and classes.).
+        """
+
+
+@Length.subaspect
+class LineLength:
+    """Number of character found in a line of code.
+    """
+    class docs:
+        example = """
+        print('The length of this line is 38')
+        """
+        example_langague = 'python'
+        importance_reason = """
+        Too long lines make code very difficult to read and maintain.
+        """
+        fix_suggestions = """
+        Splitting long lines of code into multiple shorter lines whenever
+        possible. Avoiding the usage of in-line language specific constructs
+        whenever they result in too long lines.
+        """
+    max_line_length = Taste[int](
+        'Maximum number of character for a line.',
+        (79, 80, 100, 120, 160), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """Number of lines found in a file.
+    """
+    class docs:
+        example = """
+        # This file would be a large file if we assume that the max number of
+        # lines per file is 10
+
+        class Node:
+            def __init__(self, value, left_most_child, left_sibling):
+                self.value=value
+                self.left_most_child=left_most_child
+                self.left_sibling=left_sibling
+
+        # This is example is just showing what this aspect is about, because
+        # the max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Splitting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """All whitespace found between non-whitespace characters.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including
+        # unnecessary blank lines and missing space around operators.
+
+
+
+        def func(   ):
+           return      37*-+2
+        """
+        example_language = 'python'
+        importance_reason = """
+        Useless spacing affects the readability and maintainability of a code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """Illustrate trailing spaces.
```

Comment:
> All whitespace found after the last non-whitespace character on the line until the newline.

This is 92 chars, wihich doesnt fit in the first line summary defined by PEP 257.

In other comments you ask @Asnelchristian to discard the first line .  That is not helpful.

I do agree with `Illustrate ..` is not the best, and can be improved, but a multiline summary is worse as machines cant *use* the first line if it is going to be variable length.  Machines already use the first line of a class docstring for many technical purposes.  I expect that this will be even more important for aspect classes, as machines are going to be using these strings frequently to communicate with users.